rollingpin/status.py

```diff
@@ -64,20 +64,14 @@ def fetch_deploy_status(config):
 
     fetch_req = _do_status_http_request(harold_base_url, harold_secret, salon)
 
-    # give the request a few seconds and bail out if it takes too long
-    timeout = reactor.callLater(TIMEOUT_SECONDS, fetch_req.cancel)
+    def log_timeout(d, timeout):
+        logging.warning("failed to fetch deploy status: timed out")
+        return SAFE_DEFAULT
 
-    def cancel_timeout(passthrough):
-        if timeout.active():
-            timeout.cancel()
-        return passthrough
-    fetch_req.addBoth(cancel_timeout)
+    fetch_req.addTimeout(TIMEOUT_SECONDS, reactor, onTimeoutCancel=log_timeout)
```

Comment:
switched this to use `addTimeout`, I _assume_ that that function got added in a later version of twisted, but lmk if there's a reason to avoid it.

Reply:
yup, as of 16.5. should help with any other places we need timeouts too.